File: src/common/py/data/entities/project/project_job_utils.py
from .project_job import ProjectJob, ProjectJobID


def combine_project_job_id(job: ProjectJob) -> ProjectJobID:
    """
    Combines the project job keys (project ID + connector instance) to a tuple.

    Args:
        job: The project job.

    Returns:
        The ID tuple.
    """
    return job.project_id, job.connector_instance

File: src/common/web/data/entities/project/ProjectJobUtils.ts
import { Project } from "./Project";
import { ProjectJob } from "./ProjectJob";

/**
 * Finds all active jobs of a specific project.
 *
 * @param project - The project.
 * @param jobs - All active jobs.
 *
 * @returns - All active jobs of the project.
 */
export function findProjectJobs(project: Project, jobs: ProjectJob[]): ProjectJob[] {
    return jobs.filter((job) => job.project_id == project.project_id);
}
